modules/3d/src/pointcloud/load_point_cloud.cpp

```diff
@@ -210,32 +210,32 @@ void loadMesh(const String &filename, OutputArray vertices, OutputArrayOfArrays
 
     if (texCoords.needed())
     {
-        int ch = texCoords.empty() ? 0 : texCoords.channels();
-        Mat texMat = Mat(1, static_cast<int>(vec_texCoords.size()), CV_MAKETYPE(CV_32F, nTexCoords), vec_texCoords.data());
-        if (ch == nTexCoords)
-        {
-            texMat.copyTo(texCoords);
-        }
-        else
+        if (nTexCoords)
         {
-            Mat newTexMat;
+            CV_Assert(!texCoords.fixedType() || (texCoords.type() == CV_MAKE_TYPE(CV_32F, nTexCoords)));
+
+            Mat tex3 = Mat(vec_texCoords);
             std::vector<Mat> varr;
-            cv::split(texMat, varr);
-            if (ch == 2 && nTexCoords == 3)
-            {
-                std::vector<Mat> marr = { varr[0], varr[1] };
-                cv::merge(marr, newTexMat);
-            }
-            else if (ch == 3 && nTexCoords == 2)
+            cv::split(tex3, varr);
+            std::vector<Mat> marr = { varr[0], varr[1] };
+
+            int ch = texCoords.channels();
+            if (nTexCoords == 3)
             {
-                std::vector<Mat> marr = { varr[0], varr[1], Mat::zeros(varr[0].size(), CV_32F) };
-                cv::merge(marr, newTexMat);
+                marr.push_back(varr[2]);
             }
-            else
+            else if (nTexCoords == 2)
             {
-                newTexMat = texMat;
+                if (ch == 3)
+                {
+                    marr.push_back(Mat::zeros(varr[0].size(), CV_32F));
+                }
             }
-            newTexMat.copyTo(texCoords);
+            cv::merge(marr, texCoords);
```

Comment:
Looks like mixChannels is enough instead of all the changes: https://docs.opencv.org/4.x/d2/de8/group__core__array.html#ga83b61f9875e6618ee08487b21ec71ab1. As better alternative I propose to use OutputArray/cv::Mat in readData and handle it on data creation.

Reply:
Sure, I forgot about this function.
It helped me make my code cleaner in fact, thank you